tools/massive_change.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python
+#  -*- coding: utf-8 -*-
+# License AGPLv3 (http://www.gnu.org/licenses/agpl-3.0-standalone.html)
+"""
+This script helps to make a massive change across a whole organization
+on contents of a file.
+
+Installation
+============
+
+For using this utility, you need to install these dependencies:
+
+* github3.py library for handling Github calls. To install it, use:
+  `sudo pip install github3.py`.
+
+Configuration
+=============
+
+You must have a file called oca.cfg on the same folder of the script for
+storing credentials parameters. You can generate an skeleton config running
+this script for a first time.
+
+Usage
+=====
+oca-massive-change [-h] [-p PROJECTS [PROJECTS ...]] [-e EMAIL]
+                   [-t TARGET_ORG] [-m COMMIT_MESSAGE]
+                   target path source_string target_string
+
+Performs a massive change over a whole organization.
+
+positional arguments:
+  target                Target branch
+  path                  File path
+  source_string         Source string (or regexp)
+  target_string         Target string (or regexp)
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p PROJECTS [PROJECTS ...], --projects PROJECTS [PROJECTS ...]
+                        List of specific projects to migrate
+  -e EMAIL, --email EMAIL
+                        Provides an email address used to commit on GitHub if
+                        the one associated to the GitHub account is not public
+  -t TARGET_ORG, --target-org TARGET_ORG
+                        By default, the GitHub organization used is OCA. This
+                        arg lets you provide an alternative organization
+  -m COMMIT_MESSAGE, --message COMMIT_MESSAGE
+                        By default, '[UPD] Massive change [skip ci]', but can
+                        be provided with an alternative one
+
+This script will perform the following operations for each project:
+
+* Look for the selected file in the target branch.
+* If found, modify it searching for the source string and replacing it by the
+  target string.
+* Commit the changes with optional commit message.
+
+Credits
+=======
+
+Contributors
+------------
+
+* Tecnativa - Pedro M. Baeza
+
+Maintainer
+----------
+
+.. image:: https://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit https://odoo-community.org.
+"""
+
+from __future__ import print_function
+import argparse
+import re
+from . import github_login
+from . import oca_projects
+from .config import read_config
+from github3.exceptions import NotFoundError
+
+DEFAULT_MESSAGE = "[UPD] Massive change [skip ci]"
```

Comment:
Why skip ci? It'd be helpful to know if this commit is guilty for breaking some CI, right?

Reply:
By default it should skip it before the change probably won't affect. If you don't like it, just use another, but I prefer this sane default.